This patch is the combination of 

commit c8a0ab2c9e84fcb5ae87e4646fa487c952d6b483
Author: ihayvuk <ihayvuk>
Date:   Tue Apr 20 11:26:14 2010 +0000

    Fixed bug #1517

commit af88516244edadb5a9a4cd4ae70fead4f58775c3
Author: ihayvuk <ihayvuk>
Date:   Wed Aug 18 10:18:39 2010 +0000

    Refixed bug #1517 (LabelGadget draws over parent window with the background color)

diff -up openmotif-2.3.3/lib/Xm/LabelG.c.motifzone_1517 openmotif-2.3.3/lib/Xm/LabelG.c
--- openmotif-2.3.3/lib/Xm/LabelG.c.motifzone_1517	2009-10-22 12:59:53.000000000 +0200
+++ openmotif-2.3.3/lib/Xm/LabelG.c	2011-08-01 15:24:16.793211083 +0200
@@ -47,6 +47,7 @@ static char rcsid[] = "$TOG: LabelG.c /m
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
+#define FIX_1517
 
 #include <string.h>
 #include <stdio.h>
@@ -2217,9 +2218,7 @@ LRectangle *background_box)
     switch (lw->label.fill_bg_box)
     {
         case _XmPLAIN_BG_BOX:
-#ifndef USE_XFT
             return;
-#endif
 
         case _XmFILL_BG_BOX:
         case _XmALWAYS_FILL_BG_BOX:
@@ -2336,6 +2335,30 @@ LRectangle *background_box)
     } else
     XSetClipMask (XtDisplay (lw), clipgc, None);
 
+#ifdef FIX_1517
+#ifdef USE_XFT
+    {
+    int width, height;
+    
+    int x = lw->rectangle.x + LabG_TextRect(lw).x + LabG_StringRect(lw).x;
+    int y = lw->rectangle.y + LabG_TextRect(lw).y + LabG_StringRect(lw).y;
+    
+    if (LabG_StringRect(lw).width < availW - marginal_width)
+    	width = LabG_StringRect(lw).width;
+    else
+    	width = availW - marginal_width - x;
+
+    if (LabG_StringRect(lw).height < availH - marginal_height)
+    	height = LabG_StringRect(lw).height;
+    else
+    	height = availH - marginal_height - y;
+    
+    XFillRectangle(XtDisplay(lw), XtWindow(lw), LabG_BackgroundGC(lw),
+		x, y, width, height);
+    }
+#endif
+#endif
+
     /*  Draw the pixmap or text  */
     LabelDrawBackground((Widget)lw, event, region, background_box);
 
